clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -230,24 +259,24 @@ func (s *serviceCreationLatencyMeasurement) deleteObject(svc *corev1.Service) er
 }
 
 func (s *serviceCreationLatencyMeasurement) updateObject(svc *corev1.Service) error {
-	// This measurement only works for services with LoadBalancer type.
-	if svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+	// This measurement only works for services with ClusterIP, NodePort and LoadBalancer type.
+	if svc.Spec.Type != corev1.ServiceTypeClusterIP && svc.Spec.Type != corev1.ServiceTypeNodePort && svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
 		return nil
 	}
 	key, err := runtimeobjects.CreateMetaNamespaceKey(svc)
 	if err != nil {
 		return fmt.Errorf("meta key created error: %v", err)
 	}
-	if _, exists := s.creationTimes.Get(key, creatingPhase); !exists {
-		s.creationTimes.Set(key, creatingPhase, svc.CreationTimestamp.Time)
+	if _, exists := s.creationTimes.Get(key, phaseName(creatingPhase, svc.Spec.Type)); !exists {
+		s.creationTimes.Set(key, phaseName(creatingPhase, svc.Spec.Type), svc.CreationTimestamp.Time)
 	}
 	if len(svc.Status.LoadBalancer.Ingress) < 1 {
 		return nil
 	}
-	if _, exists := s.creationTimes.Get(key, ipAssigningPhase); exists {
+	if _, exists := s.creationTimes.Get(key, phaseName(ipAssigningPhase, svc.Spec.Type)); exists {
```

Comment:
This no longer works - services of type ClusterIp and NodePort doesn't have len(svc.Status.LoadBalancer.Ingress) >= 1

Reply:
fixed